src/devices/KeyMatrix/KeyMatrix.cs

```diff
@@ -0,0 +1,211 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.KeyMatrix
+{
+    /// <summary>
+    /// GPIO key matrix Driver
+    /// </summary>
+    public class KeyMatrix : IDisposable
+    {
+        /// <summary>
+        /// Get output pins
+        /// </summary>
+        public IEnumerable<int> OutputPins => _outputPins;
+
+        /// <summary>
+        /// Get input pins
+        /// </summary>
+        public IEnumerable<int> InputPins => _inputPins;
+
+        /// <summary>
+        /// Get all buttons' values
+        /// </summary>
+        public ReadOnlySpan<PinValue> Values => _buttonValues.AsSpan();
+
+        /// <summary>
+        /// Get or set interval in milliseconds
+        /// </summary>
+        public int ScanInterval { get; set; }
+
+        /// <summary>
+        /// Delegate of key matrix event
+        /// </summary>
+        public delegate void PinChangeEventHandler(object sender, KeyMatrixEventArgs pinValueChangedEventArgs);
+
+        private int[] _outputPins;
+        private int[] _inputPins;
+        private GpioController _masterGpioController;
+        private PinValue[] _buttonValues;
+        private bool _pinsOpened;
+
+        /// <summary>
+        /// Initialize key matrix
+        /// </summary>
+        /// <param name="masterController">GPIO controller</param>
+        /// <param name="outputPins">Output pins</param>
+        /// <param name="inputPins">Input pins</param>
+        /// <param name="scanInterval">Scanning interval in milliseconds</param>
+        public KeyMatrix(GpioController masterController, IEnumerable<int> outputPins, IEnumerable<int> inputPins, int scanInterval)
```

Comment:
assuming this is the only controller being used I'd name this `gpioController` or just `controller`